### PR TITLE
Refactoring ONe compute plugin

### PR DESCRIPTION
**Pull request by pauloewerton**
This change adds:

- Datastore is chosen by listing available options in order to avoid having to put a datastore id in cloud.conf;
- Userdata is created using an ONe-compatible launch command generator and put on the right request parameter;
- Public key is put on the right request parameter;
- Disk size conversions, since ONe handles sizes in megabytes not gigabytes as expected by Fogbow.

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePlugin.java [hunk 1]
```diff
@@ -86,18 +77,21 @@ public boolean hasFailed(String cloudState) {
 
 	@Override
 	public String requestInstance(ComputeOrder computeOrder, CloudUser cloudUser) throws FogbowException {
```
Comment (raonismaneoto) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePlugin.java]: you could extract the request creation to an aux method in order to decrease the main method's size. Also, we're trying to follow a pattern where we create another aux method usually named as do**Request. This method wraps some operations responsible for really request the instance, in this case, and we just return its call...

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/OpenNebulaLaunchCommandGenerator.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public String createLaunchCommand(ComputeOrder order) {
+        List<UserData> userDataScripts = order.getUserData();
+        StringBuilder userDataBuilder = new StringBuilder();
+        Scanner scanner = null;
+
+        try {
+            scanner = new Scanner(new File(ONE_STARTUP_SCRIPT_FILE_PATH));
+            userDataBuilder.append(scanner.useDelimiter("\\A").next());
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/OpenNebulaLaunchCommandGenerator.java]: sure

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/OpenNebulaLaunchCommandGenerator.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        if (userDataScripts != null) {
+            for (UserData userDataScript : userDataScripts) {
+                if (userDataScript != null) {
+                    String normalizedExtraUserData = null;
+                    String extraUserDataFileContent = userDataScript.getExtraUserDataFileContent();
+                    // NOTE(pauloewerton): since only one script can be added to the request, we're appending only
+                    // the bash scripts to the default script in order to prevent errors
+                    if (extraUserDataFileContent != null &&
+                            userDataScript.getExtraUserDataFileType() == CloudInitUserDataBuilder.FileType.SHELL_SCRIPT) {
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/OpenNebulaLaunchCommandGenerator.java]: good catch

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/CreateComputeRequest.java [hunk 4]
```diff
@@ -66,34 +66,30 @@ private Disk buildDisk(Builder builder) {
...
-		context.setEncoding(builder.contextEncoding);
-		context.setUserdata(builder.contextUserdata);
+		context.setPublicKey(builder.publicKey);
+		context.setUserName(builder.userName);
+		context.setStartScriptBase64(builder.startScriptBase64);
 		context.setNetwork(builder.contextNetwork);
 		return context;
 	}
 
 	public static class Builder {
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/CreateComputeRequest.java]: agreed

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePlugin.java (hunk 1 above)
Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePlugin.java]: thanks for the heads up. will do

**Comment by pauloewerton** [comment]
@raonismaneoto thanks for the reply. i'll address your comments in the next commit

**Comment by raonismaneoto** [comment]
good job @pauloewerton

**Pull request merged**